toolboxes/feel/feelmodels/hdg/mixedpoissoncreate.cpp

```diff
@@ -0,0 +1,294 @@
+            }
+
+            itType = mapField.find( "InterfaceCondition" );
+            if ( itType != mapField.end() )
+            {
+                Feel::cout << "Interface condition:";
+                for ( auto const& exAtMarker : (*itType).second )
+                {
+                    std::string marker = exAtMarker.marker();
+                    if ( M_mesh->hasFaceMarker(marker) )
+                        Feel::cout << " " << marker;
+                    else
+                        Feel::cout << std::endl << "WARNING!! marker " << marker << "does not exist!" << std::endl;
+                }
+                Feel::cout << std::endl;
+            }
+        }
+    }
+
+    if ( M_IBCList.empty() )
+        M_integralCondition = 0;
+    else
+        M_integralCondition = M_IBCList.size();
+
+    if ( boost::icontains(modelProperties().models().model().equations(),"picard") )
+        M_isPicard = true;
+    else
+        M_isPicard = false;
+}
+
+MIXEDPOISSON_CLASS_TEMPLATE_DECLARATIONS
+void
+MIXEDPOISSON_CLASS_TEMPLATE_TYPE::initSpaces()
+{
+    // Mh only on the faces whitout integral condition
+    auto complement_integral_bdy = complement(faces(M_mesh),[this]( auto const& ewrap ) {
+            auto const& e = unwrap_ref( ewrap );
+            for( auto exAtMarker : this->M_IBCList)
+            {
+                if ( e.hasMarker() && e.marker().value() == this->M_mesh->markerName( exAtMarker.marker() ) )
+                    return true;
+            }
+            return false; });
+
+    auto face_mesh = createSubmesh( M_mesh, complement_integral_bdy, EXTRACTION_KEEP_MESH_RELATION, 0 );
+
+
+    M_Vh = Pdhv<Order>( M_mesh, true);
+    M_Wh = Pdh<Order>( M_mesh, true );
+    M_Mh = Pdh<Order>( face_mesh, true );
+    // M_Ch = Pch<0>( M_mesh, true );
+    M_M0h = Pdh<0>( face_mesh );
+
+	std::vector<std::string> ibc_markers(M_integralCondition);
+	for( int i = 0; i < M_integralCondition; i++)
+	{
+		ibc_markers.push_back(M_IBCList[i].marker());
+	}
+
+    auto ibc_mesh = createSubmesh( M_mesh, markedfaces(M_mesh, ibc_markers), EXTRACTION_KEEP_MESH_RELATION, 0 );
+    M_Ch = Pch<0>( ibc_mesh, true );
+
+    Feel::cout << "Vh<" << Order << "> : " << M_Vh->nDof() << std::endl
+         << "Wh<" << Order << "> : " << M_Wh->nDof() << std::endl
+         << "Mh<" << Order << "> : " << M_Mh->nDof() << std::endl;
+    if ( M_integralCondition )
+        Feel::cout << "Ch<" << 0 << "> : " << M_Ch->nDof() << std::endl;
+
+    auto ibcSpaces = std::make_shared<ProductSpace<Ch_ptr_t,true> >( M_integralCondition, M_Ch);
+    M_ps = std::make_shared<product2_space_type>(product2(ibcSpaces,M_Vh,M_Wh,M_Mh));
+
+    M_up = M_Vh->element( "u" );
+    M_pp = M_Wh->element( "p" );
+
+   	for( int i = 0; i < M_integralCondition; i++ )
+       	M_mup.push_back(M_Ch->element("mup"));
+
+	solve::strategy s = boption(prefixvm(prefix(), "use-sc"))?solve::strategy::static_condensation:solve::strategy::monolithic;
```

Comment:
option should be set up in constructor and then only use API to decide what to do. Otherwise, the strategies can only be decided by changing the command line. we pon't want this